File: backend/app/schemas/metadata.py
from datetime import date

from pydantic import BaseModel


class MetadataSearchResult(BaseModel):
    provider: str
    provider_id: str
    title: str
    summary: str | None = None
    release_date: date | None = None
    score: float | None = None

class MetadataDetails(BaseModel):
    provider: str
    provider_id: str
    title: str
    description: str | None = None
    release_date: date | None = None
    genres: list[str] = []
    developers: list[str] = []
    publishers: list[str] = []
    tags: list[str] = []
    cover_urls: list[str] = []
    banner_urls: list[str] = []
    logo_urls: list[str] = []
    artwork_urls: list[str] = []

File: frontend/lib/types/metadata.ts
export interface MetadataReviewItem {
  id: string;
  entry_id: string;
  notes?: string;
  suggested_title?: string;
  created_at?: string;
}

export interface MetadataSearchResult {
  provider: string;
  provider_id: string;
  title: string;
  summary?: string | null;
  release_date?: string | null;
  score?: number | null;
}

export interface MetadataDetails {
  provider: string;
  provider_id: string;
  title: string;
  description?: string | null;
  release_date?: string | null;
  genres: string[];
  developers: string[];
  publishers: string[];
  tags: string[];
  cover_urls: string[];
  banner_urls: string[];
  logo_urls: string[];
  artwork_urls: string[];
}
